sceptre/template_handlers/formatted_yaml.py

```diff
@@ -0,0 +1,68 @@
+"""
+This module contains a template handler that can load files from disk and
+format the content.
+"""
+import re
+import os
+from sceptre.template_handlers.file import File
+
+
+class FormattedYaml(File):
+    """
+    Template handler that can load files from disk.
+    Supports JSON, YAML, Jinja2 and Python.
+
+    This class extends the `File` class from the
+    `sceptre.template_handlers.file` module. It provides additional
+    functionality to format YAML content by removing trailing whitespaces
+    and blank lines, adding line breaks before lines that don't start with
+    spaces, tabs, or specific prefixes, and adding a newline at the end
+    of the file.
+    """
+
+    # Class attribute for line start exceptions
+    no_newline_prefixes = ["---", "AWSTemplateFormatVersion", "Description"]
+
+    def handle(self):
+        """
+        Handles the template by returning the template contents.
+
+        Returns:
+            str or bytes: The template contents.
+        """
+        contents = super().handle()
+        if isinstance(contents, str):
+            return self._format_yaml_content(contents)
```

Comment:
shouldn't this formatting only happen for jinja based templates?

Reply:
I guess it should happen for anyone who chooses to call the `formatted_yaml` template handler. The intended use case is Jinja-based templates although in theory it could work on any YAML templates. I agree though that some more sanity-checking here would not hurt. I'll have a think.